### ISSUE TITLE
template and copy modules should have state parameter

### ISSUE THREAD

**Issue by Vladimir-csp** [ansible/ansible-modules-core#3934]
##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

copy
template
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

As was suggested in https://github.com/ansible/ansible/issues/6929, I am opening new issue on this matter
Please add 'state' parameter for template and copy modules, so items could be removed with the same task by using condition or variable override.
##### STEPS TO REPRODUCE

```

```

##### EXPECTED RESULTS

Something like this should be possible

```
- name: govern foo template
  template: src=foo dest=/etc/foo state="{% if enable_foo == true %}present{% else %}absent{% endif%}"
```

On second thought, file, copy, and template modules could be merged together into file module to make some cool things like:

```
- name: linked template
  file: template=foo.conf state=link src=../bar/bar.conf dest=/etc/foo/foo.conf
# makes link /etc/foo/foo.conf > /etc/bar/bar.conf,
# templates  /etc/foo/foo.conf  from foo.conf effectively writing to  /etc/bar/bar.conf.
```
##### ACTUAL RESULTS

```

```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29781
